python/ray/util/sgd/v2/migration/sgd-migration-guide.md

```diff
@@ -0,0 +1,224 @@
+# Ray SGD: v1 to v2 migration guide
+
+In Ray 1.7, we are rolling out a new and more streamlined version of Ray SGD. Ray SGD v2 focuses on usability and composability- it has a much simpler API, has support for more deep learning backends, integrates better with other libraries in the Ray ecosystem, and will continue to be actively developed with more features. 
+
+This guide will cover to easily migrate existing code from Ray SGD v1 to Ray SGD v2. If you are new to Ray SGD as a whole, you should get started with Ray SGD v2 directly: https://docs.ray.io/en/master/raysgd/v2/raysgd.html. 
+
+For a full list of features that Ray SGD v2 provides, please check out the user guide: https://docs.ray.io/en/master/raysgd/v2/user_guide.html. 
+
+If there are any issues or anything missing with this guide or any feedback on Ray SGD v2 overall, please file a Github issue on the Ray repo: https://github.com/ray-project/ray/issues!
+
+## What are the API differences?
+There are 2 primary API differences between Ray SGD v1 and v2.
+
+1. There is a single `Trainer` interface for all backends (torch, tensorflow, horovod), and the backend is simply specified via an argument: `Trainer(backend="torch")`, `Trainer(backend="horovod")`, etc. Any features that we add to Ray SGD will be supported for all backends, and there won't be any API divergence like there was with a separate `TorchTrainer` and `TFTrainer`.
+2. The `TrainingOperator` and creator functions are replaced by a generic user-defined training function. You no longer have to make your training logic fit into a restrictive interface. In Ray SGD v2, you simply have to provide a training function that describes the full logic for your training execution and this will be distributed by Ray SGD. 
+```python
+# Torch Example
+def train_func_distributed():
+    num_epochs = 3
+    model = NeuralNetwork()
+    model = DistributedDataParallel(model)
+    loss_fn = nn.MSELoss()
+    optimizer = optim.SGD(model.parameters(), lr=0.1)
+
+    for epoch in range(num_epochs):
+        output = model(input)
+        loss = loss_fn(output, labels)
+        optimizer.zero_grad()
+        loss.backward()
+        optimizer.step()
+        print(f"epoch: {epoch}, loss: {loss.item()}")
+
+from ray.sgd import Trainer
+
+trainer = Trainer(backend="torch", num_workers=4)
+trainer.start()
+results = trainer.run(train_func_distributed)
+trainer.shutdown()
+```
+Currently, this means that the user is now responsible for modifying their code to support distributed training (specifying `DistributedDataParallel` for `torch` or `MultiWorkerMirroredStrategy` for `tensorflow`) as opposed to having this be automatically handled internally. However, we have plans to provide utilities that you can use to automatically handle these recipes for you.
+
```

Comment:
It would be nice to end this section with a quick list of what's upcoming in the doc so the user knows what to expect. I found myself asking "what about reporting or checkpointing" while reading the `Training Logic` section, which intentionally covers just the pure training aspect.

Example:
```
In the following sections, we will guide you through the steps to migrate:
1. Training Logic
2. Interacting with Trainer state (intermediate metrics, checkpointing)
3. Hyperparameter Tuning with Ray Tune
```

Reply:
Added!